### PR TITLE
[stable/datadog] setting default daemonset updateStrategy to RollingUpdate

### PR DESCRIPTION
**Pull request by max-rocket-internet**
#### What this PR does / why we need it:

Sets default `updateStrategy` to `RollingUpdate` for the daemonset. Reading through the `values.yaml` it shows this is the default anyway but in `daemonset.yaml` uses `OnDelete`. I think 99% of the time, people want `RollingUpdate` though, otherwise you have to manually go and delete the pods.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by irabinovitch** [comment]
@max-rocket-internet Thanks for the PR. We're discussing internally and should have an update for you shortly.

**Review by hkaj — APPROVED** [review]
Thanks @max-rocket-internet 
I see you bumped the version waiting for your other PR to be merged 👀 
This one looks good to me, waiting for the other one to get it before merging.

**Comment by max-rocket-internet** [comment]
@hkaj can we merge this now that the other PR is merged?

**Comment by hkaj** [comment]
Thanks @max-rocket-internet !

**Pull request merged**
